src/ApplicationInsights.Kubernetes/TelemetryInitializers/SimplePerformanceCounterTelemetryInitializer.cs

```diff
@@ -0,0 +1,69 @@
+#if NETSTANDARD2_0
+
+using System;
+using System.Diagnostics;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.ApplicationInsights.Extensibility;
+using static Microsoft.ApplicationInsights.Kubernetes.TelemetryProperty;
+
+namespace Microsoft.ApplicationInsights.Kubernetes
+{
+    internal class SimplePerformanceCounterTelemetryInitializer : ITelemetryInitializer
+    {
+        TimeSpan _lastCPUSample;
+        Stopwatch _cpuWatch = new Stopwatch();
+
+        public void Initialize(ITelemetry telemetry)
+        {
+            if (telemetry is ISupportMetrics metricsTelemetry)
+            {
+                // Write CPU metrics
+                TimeSpan endCPUTime = GetCPUTime();
+                if (_lastCPUSample != default)
+                {
+                    if (_cpuWatch.ElapsedMilliseconds > 0)
+                    {
+                        int processorCount = Environment.ProcessorCount;
```

Comment:
This does not seem to be used in the CPU time calculation ☺️ Also the value could probably be cached

Reply:
Good catch. It had been missed...

Reply:
Fixed in the latest code.